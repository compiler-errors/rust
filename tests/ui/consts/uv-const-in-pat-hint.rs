//@ revisions: stock gce
//@ check-pass

#![feature(associated_const_equality)]
#![cfg_attr(gce, feature(generic_const_exprs))]
//[gce]~^ WARN the feature `generic_const_exprs` is incomplete

fn main() {
    // Make sure we don't ICE in THIR pattern analysis when writeback
    // may, depending on GCE's lazy normalization of consts, contain
    // an unevaluated const.
    let [(), ()]: [(); 1 + 1] = [(), ()];
}