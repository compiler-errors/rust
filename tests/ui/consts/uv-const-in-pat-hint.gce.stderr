warning: the feature `generic_const_exprs` is incomplete and may not be safe to use and/or cause compiler crashes
  --> $DIR/uv-const-in-pat-hint.rs:5:26
   |
LL | #![cfg_attr(gce, feature(generic_const_exprs))]
   |                          ^^^^^^^^^^^^^^^^^^^
   |
   = note: see issue #76560 <https://github.com/rust-lang/rust/issues/76560> for more information
   = note: `#[warn(incomplete_features)]` on by default

warning: 1 warning emitted

